exonum/src/events/noise/tests.rs

```diff
@@ -107,6 +112,84 @@ fn test_converted_keys_handshake() {
         .expect("Unable to transition session into transport mode");
 }
 
+#[test]
+fn test_noise_encrypt_decrypt_short_message() {
+    test_encrypt_decrypt(64);
+}
+
+#[test]
+fn test_noise_encrypt_decrypt_long_message() {
+    test_encrypt_decrypt(MAX_MESSAGE_LENGTH + 1);
+}
+
+#[test]
+fn test_noise_encrypt_decrypt_bogus_message() {
+    let msg_size = 64;
+
+    let (mut h_i, mut h_r) = create_noise_sessions();
+    let mut buffer_msg = BytesMut::with_capacity(msg_size);
+
+    let _res = h_i.encrypt_msg(&vec![0_u8; msg_size], &mut buffer_msg);
+
+    let len = LittleEndian::read_u32(&buffer_msg[..HEADER_LENGTH]) as usize;
+    let mut to_decrypt = vec![];
+    to_decrypt.extend_from_slice(&buffer_msg);
+
+    // Wrong length.
+    let res = h_r.decrypt_msg(len - 1, &mut BytesMut::from(&to_decrypt[..]));
```

Comment:
Because it mutates noise `session` in `NoiseWrapper::read` method while reading the decrypting the message. But I agree to move chunking logic to codec, maybe not in this PR.